src/dune_engine/scheduler.ml

```diff
@@ -782,17 +782,44 @@ end = struct
   ;;
 end
 
+module Trigger : sig
+  (** Conceptually, a [unit Fiber.Ivar.t] that may be filled more than once. *)
+  type t
+
+  val create : unit -> t
+  val trigger : t -> Fiber.fill list
```

Comment:
How about returning an option? Would make the interface more precise.

Reply:
I wouldn't say using an option is more precise. Option and list communicate different things. In this case, the user shouldn't care whether zero, one, or many fills in the list; all it is supposed to do is just forward it along to the fiber scheduler. If it were an option, we would have to convert it to a list later.

Reply:
Why would we convert it to a list? We'd just cons it to an existing list if something was there?

This issue isn't crucial but I'm not convinced by your argument for using a list. If the user shouldn't care about zero, one or many fills, then returning the list goes against your intention. You're now making it possible for the user to discriminate multiple fills from a single one. To actually get to your stated goal, you would need to introduce some abstract type that only allows for appending fills. Now I don't see that as needed because the code is trivial, but returning a richer type to the user gives more rope to hang himself, not less.

Reply:
EDIT: Oh, I see you resolved this comment. Feel free to ignore below, but I'm posting it in cases it's enough to convince you. :)

`option` says "there is one fill or there is no fill." `list` says "I won't tell
you how fills there are; be prepared for any amount." In that sense, `list` is a
bit more abstract because the user gets less information about the data being
returned.

It's true that now `list` gives the user more freedom to misunderstand the argument, but it would look pretty weird if they did, and there is only one callsite, so I'm not too concerned about that.

Perhaps the main reason I picked `list` is that it has a convenient append operation. Unless I've misunderstood the suggested change, using an `option` would require a match expression when calling `trigger`; it's not so bad, but it seems a bit noisy to me.